diff --git a/pybind11/commands.py b/pybind11/commands.py
index fa7eac3..7cf4a41 100644
--- a/pybind11/commands.py
+++ b/pybind11/commands.py
@@ -6,9 +6,7 @@ DIR = os.path.abspath(os.path.dirname(__file__))
 
 
 def get_include(user=False):
-    installed_path = os.path.join(DIR, "include")
-    source_path = os.path.join(os.path.dirname(DIR), "include")
-    return installed_path if os.path.exists(installed_path) else source_path
+    return "@include@"
 
 
 def get_cmake_dir():
