diff --git a/tools/pybind11Tools.cmake b/tools/pybind11Tools.cmake
index 49876eb..2fd39e2 100644
--- a/tools/pybind11Tools.cmake
+++ b/tools/pybind11Tools.cmake
@@ -32,15 +32,15 @@ include(CheckCXXCompilerFlag)
 include(CMakeParseArguments)
 
 # Warn or error if old variable name used
-if(PYBIND11_CPP_STANDARD)
+if(CMAKE_CXX_STANDARD)
   if(NOT CMAKE_CXX_STANDARD)
-    string(REGEX MATCH [=[..^]=] VAL "${PYBIND11_CPP_STANDARD}")
+    string(REGEX MATCH [=[..^]=] VAL "${CMAKE_CXX_STANDARD}")
     set(supported_standards 11 14 17 20)
     if("${VAL}" IN_LIST supported_standards)
       message(WARNING "USE -DCMAKE_CXX_STANDARD=${VAL} instead of PYBIND11_PYTHON_VERSION")
       set(CMAKE_CXX_STANDARD ${VAL})
     else()
-      message(FATAL_ERROR "PYBIND11_CPP_STANDARD should be replaced with CMAKE_CXX_STANDARD")
+      message(FATAL_ERROR "CMAKE_CXX_STANDARD should be replaced with CMAKE_CXX_STANDARD")
     endif()
   endif()
 endif()
